src/js/model/racingcar.js

```diff
@@ -1,39 +1,79 @@
 import { getRandomNumber } from '../utils/random.js';
-import {
-  TRIM_BETWEEN_COMMA,
-  GENERATION_MIN,
-  GENERATION_MAX,
-  GO_OR_STOP_CONDITION,
-  SEPARATOR,
-} from '../constant/racingcar.js';
+import { NAME_LENGTH_MIN, NAME_LENGTH_MAX } from '../constant/racingcar.js';
+import { catchMessage, getType } from '../validate/validate.js';
+import ERROR_MESSAGES from '../constant/errorMessages.js';
 
 const racingManager = {
   names: [],
   trialCount: 0,
   gameResult: [],
 
+  resetAll() {
+    this.names = [];
+    this.trialCount = 0;
+    this.gameResult = [];
+  },
+
   trimNames(value) {
-    return value.replace(TRIM_BETWEEN_COMMA, SEPARATOR).trim();
+    return value.replace(/\s*,\s*/g, ',').trim();
   },
 
   splitName(name) {
-    return name.split(SEPARATOR);
+    return name.split(',');
   },
 
   isGoOrStop(randomNum) {
-    return randomNum > GO_OR_STOP_CONDITION;
+    return randomNum > 3;
+  },
+
+  goCondition() {
+    const randomNum = getRandomNumber(0, 9);
+    return this.isGoOrStop(randomNum);
   },
```

Comment:
```suggestion
  goCondition(gameStrategy) {
    return gameStrategy();
  },
```

이렇게 자동차의 전진 조건을 아예 주입 받는다고 가정해보면 어떨까요?

이럴 경우 racingManager는 gameStrategy의 interface만 알고 있다면 테스트하기 쉬운 코드가 되지 않을까요?